server/controllers/finance/reports/income_expense/index.js

```diff
@@ -89,10 +89,29 @@ function document(req, res, next) {
 
       const profits = [];
       const losses = [];
+      const totals = {
+        income : { balance : 0, previousBalance : 0 },
+        expense : { balance : 0, previousBalance : 0 },
+      };
 
       tree.walk(node => profits.push(node), true, income);
+      // sum income
+      tree.walk(node => {
```

Comment:
This code is correct, but the solution is even easier!  The `income` here is the root of the `income` account tree, and `expense` is the root of the expense tree.  Every node in the tree has the following properties: `balance`, `previousBalance`, `difference`, `variance`.  They are computed [here](
https://github.com/IMA-WorldHealth/bhima-2.X/pull/2804/files#diff-950ab90f8fd78e6083caa9787d57fac4L219).

So the total of all income and all expense accounts are already calculated!  You can just grab them from the root node of each tree:
```js
const totals = {
  income: { balance : income.balance, previousBalance: income.previousBalance, difference: income.difference },
  expense : { balance : expense.balance, expense.previousBalance, difference: expense.difference }
};
```

Reply:
```income``` is a tree, so ```income.balance``` is not the total of all nodes